test/test_nestedtensor.py

```diff
@@ -0,0 +1,210 @@
+import torch
+import unittest
+from common_utils import TEST_WITH_NESTEDTENSORS
+import torch.tensortypes.nestedtensor.codegen as codegen
+
+if TEST_WITH_NESTEDTENSORS:
+    import torch.tensortypes.nestedtensor as nestedtensor
+    from common_utils import TestCase
+    NestedTensor = nestedtensor.NestedTensor
+else:
+    print('NestedTensors not available, skipping tests')
+    TestCase = object
+
+
+def _shape_prod(shape_):
+    shape = tuple(shape_)
+    start = 1
+    for s in shape:
+        start = start * s
+    return start
+
+# From torchaudio by jamarshon
+def random_float_tensor(seed, size, a=22695477, c=1, m=2 ** 32,
```

Comment:
Err... why not just torch.uniform?

Reply:
Reproducibility. See the other comments.

Reply:
The comment is pretty hidden in the comments of this PR, would be good to add a comment into the code here so that someone else looking through the code doesn't try to replace it with torch.uniform.

Reply:
No, I don't understand. `torch.uniform` is reproducible too, if you explicitly specify a seed. If you need to pass in the seed explicitly, well, we have a new Generator object courtesy of @syed-ahmed which will let you do just that.

Reply:
It's not invariant under the RNG and you have to set the seed (a global variable) to get the same result every time. Why does the RNG matter? Some bugs are discovered further down the road. 
I wasn't aware of the Python Generator class. Creating one with a specific seed and then passing into the random number generator might indeed be the solution, but only if the type of RNG can be fixed.

I might be a bit overly perfectionist here, but I've been in the situation before where I'm running torch.rand over and over again to reproduce some kind of bug.